src/storage/src/hummock/iterator/forward_user.rs

```diff
@@ -103,7 +104,12 @@ impl<I: HummockIterator<Direction = Forward>> UserIterator<I> {
                 // handle delete operation
                 match self.iterator.value() {
                     HummockValue::Put(val) => {
-                        if self.delete_range_aggregator.should_delete(key, epoch) {
+                        while self.delete_range_iter.is_valid()
+                            && self.delete_range_iter.next_user_key().le(key)
+                        {
+                            self.delete_range_iter.next();
+                        }
+                        if self.delete_range_iter.current_epoch() >= epoch {
```

Comment:
would you mind moving this code to one method of `delete_range_iter `?

Reply:
> would you mind moving this code to one method of `delete_range_iter `?

done